src/core/crypto.c

```diff
@@ -1687,7 +1687,7 @@ QuicCryptoProcessDataComplete(
 
     QuicCryptoValidate(Crypto);
 
-    if (!Crypto->CertValidationPending) {
+    if (!Crypto->CertValidationPending && !Crypto->TicketValidationPending) {
```

Comment:
Instead of doing it this way (for both ticket and cert validation) I think a better way is to just return at the beginning of this function `if (Crypto->CertValidationPending || Crypto->TicketValidationPending)`. Then, in the completion, just call this function (`QuicCryptoProcessDataComplete`) with the cached `RecvBufferConsumed `.